apps/dot_ext/loggers.py

```diff
@@ -142,17 +147,43 @@ def set_session_values_from_auth_flow_uuid(request, auth_flow_uuid):
             pass
 
 
-def update_instance_auth_flow_trace_with_code(auth_uuid, code):
+def set_session_auth_flow_trace_value(request, key, value):
+    '''
+    Set auth flow key value in the session.
+    '''
+    if request.session:
+        request.session[key] = value
+
+
+def update_instance_auth_flow_trace_with_code(auth_dict, code):
     '''
-    Update AuthFlowUuid instance with code value.
+    Update AuthFlowUuid instance with code, crosswalk_type and share_demographic_scopes values.
 
     CALLED FROM:  apps.dot_ext.views.authorization.AuthorizationView.form_valid()
     '''
+    auth_uuid = auth_dict.get('auth_uuid', None)
+    auth_crosswalk_type = auth_dict.get('auth_crosswalk_type', None)
+    auth_share_demographic_scopes = auth_dict.get('auth_share_demographic_scopes', None)
+
     try:
-        if code and len(code.strip()) != 0:
+        if auth_uuid:
             auth_flow_uuid = AuthFlowUuid.objects.get(auth_uuid=auth_uuid)
-            # Set code.
-            auth_flow_uuid.code = code
+        else:
+            auth_flow_uuid = None
+
+        if auth_flow_uuid:
+            if code and len(code.strip()) != 0:
+                auth_flow_uuid.code = code
+
+            if auth_crosswalk_type:
+                auth_flow_uuid.auth_crosswalk_type = auth_crosswalk_type
+
+            if auth_share_demographic_scopes:
+                if auth_share_demographic_scopes == "True":
+                    auth_flow_uuid.auth_share_demographic_scopes = True
+                elif auth_share_demographic_scopes == "False":
+                    auth_flow_uuid.auth_share_demographic_scopes = False
```

Comment:
Couldn't you move this code into the the previous if statement as `auth_flow_uuid` won't exist otherwise?

Reply:
@njdister You're right. That looks better.